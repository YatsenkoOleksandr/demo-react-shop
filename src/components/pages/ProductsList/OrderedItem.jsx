import PropTypes from 'prop-types';
import { faTrashAlt, faPlusSquare, faMinusSquare } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Button } from 'react-bootstrap';

export const OrderedItem = ({orderedProductItem, increaseCount, decreaseCount, removeFromCart}) => {
  return <div className='demo-component'>
    <div>
      <span style={{cursor: 'pointer'}}>
        <FontAwesomeIcon
          icon={faTrashAlt}
          onClick={() => removeFromCart(orderedProductItem)}
      />
      </span>
    </div>
    <div>
      <div>
        <img
          src={orderedProductItem.previewImageLink}
          alt={`${orderedProductItem.name} preview`} />
      </div>
      <div>
        <label>{orderedProductItem.name}</label>
      </div>
      <div>
        <label>{orderedProductItem.price}</label>
      </div>
      <div>
        <Button
            disabled={orderedProductItem.count < 2}
            onClick={() => decreaseCount(orderedProductItem)}>
          <FontAwesomeIcon icon={faMinusSquare} />
        </Button>
        <span>{orderedProductItem.count}</span>
        <Button onClick={() => increaseCount(orderedProductItem)}>
          <FontAwesomeIcon icon={faPlusSquare} />
        </Button>
      </div>
    </div>
  </div>
};

OrderedItem.propTypes = {
  orderedProductItem: PropTypes.object.isRequired,
  increaseCount: PropTypes.func.isRequired,
  decreaseCount: PropTypes.func.isRequired,
  removeFromCart: PropTypes.func.isRequired,
};
