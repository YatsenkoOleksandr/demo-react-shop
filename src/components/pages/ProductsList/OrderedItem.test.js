import { describe, test, expect, jest } from '@jest/globals';
import React from 'react';
import renderer from 'react-test-renderer'
import { OrderedItem } from './OrderedItem';

describe('OrderedItem', () => {
  test('should match snapshot and have disabled remove from cart button when ordered 1 item', () => {
    const orderedProductItem = {
      id: 100,
      name: 'Road Bike',
      price: 300,
      count: 1,
      previewImageLink: 'https://veloplaneta.com.ua/media/catalog/product/cache/0c35bc1a2ea936bd9c1c17139a73ce74/s/k/skd-52-93.jpg'
    };

    const decreaseCount = jest.fn();
    const increaseCount = jest.fn();
    const removeFromCart = jest.fn();

    const tree = renderer
      .create(<OrderedItem
        orderedProductItem={orderedProductItem}
        increaseCount={increaseCount}
        decreaseCount={decreaseCount}
        removeFromCart={removeFromCart} />)
      .toJSON();

      expect(tree).toMatchSnapshot();
  });

  test('should match snapshot and have enabled remove from cart button when ordered 2 items', () => {
    const orderedProductItem = {
      id: 200,
      name: 'Touring',
      price: 200,
      count: 2,
      previewImageLink: 'https://veloplaneta.com.ua/media/catalog/product/cache/0c35bc1a2ea936bd9c1c17139a73ce74/s/k/skd-62-99.jpg'
    };

    const decreaseCount = jest.fn();
    const increaseCount = jest.fn();
    const removeFromCart = jest.fn();

    const tree = renderer
      .create(<OrderedItem
        orderedProductItem={orderedProductItem}
        increaseCount={increaseCount}
        decreaseCount={decreaseCount}
        removeFromCart={removeFromCart} />)
      .toJSON();

      expect(tree).toMatchSnapshot();
  });
});
