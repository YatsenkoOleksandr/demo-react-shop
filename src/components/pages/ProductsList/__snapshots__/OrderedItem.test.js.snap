// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderedItem should match snapshot and have disabled remove from cart button when ordered 1 item 1`] = `
<div
  className="demo-component"
>
  <div>
    <span
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-trash-alt fa-w-14 "
        data-icon="trash-alt"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
  </div>
  <div>
    <div>
      <img
        alt="Road Bike preview"
        src="https://veloplaneta.com.ua/media/catalog/product/cache/0c35bc1a2ea936bd9c1c17139a73ce74/s/k/skd-52-93.jpg"
      />
    </div>
    <div>
      <label>
        Road Bike
      </label>
    </div>
    <div>
      <label>
        300
      </label>
    </div>
    <div>
      <button
        className="btn btn-primary"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-minus-square fa-w-14 "
          data-icon="minus-square"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM92 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H92z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <span>
        1
      </span>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-plus-square fa-w-14 "
          data-icon="plus-square"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-32 252c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92H92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`OrderedItem should match snapshot and have enabled remove from cart button when ordered 2 items 1`] = `
<div
  className="demo-component"
>
  <div>
    <span
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-trash-alt fa-w-14 "
        data-icon="trash-alt"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
  </div>
  <div>
    <div>
      <img
        alt="Touring preview"
        src="https://veloplaneta.com.ua/media/catalog/product/cache/0c35bc1a2ea936bd9c1c17139a73ce74/s/k/skd-62-99.jpg"
      />
    </div>
    <div>
      <label>
        Touring
      </label>
    </div>
    <div>
      <label>
        200
      </label>
    </div>
    <div>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-minus-square fa-w-14 "
          data-icon="minus-square"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM92 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H92z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <span>
        2
      </span>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-plus-square fa-w-14 "
          data-icon="plus-square"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-32 252c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92H92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
